e2e/openshift/openshift_packaging_test.go

```diff
@@ -0,0 +1,340 @@
+
+// This is the YAML used to create a subscription
+// in OpenShift to deploy the operator.
+const subscription = `apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: myoperator
+  namespace: test
+spec:
+  channel: stable
+  installPlanApproval: Automatic
+  name: cockroachdb-certified
+  source: cockroach-test
+  sourceNamespace: openshift-marketplace
+  startingCSV: cockroach-operator.{{.AppVersion}}`
+
+// YAMLTemplate struct is used to render the
+// above YAML files
+type YAMLTemplate struct {
+	AppVersion     string
+	DockerRegistry string
+}
+
+// delete is a helper variable used to delete resources
+var delete = []string{
+	"delete",
+	"-n",
+}
+
+// TestPackage is used to install the operator in an OpenShift
+// cluster. It uses three different YAML files and uses the test
+// namespace. This program uses oc and also installs example.yaml
+// to test that the operator is functioning.
+func TestPackaging(t *testing.T) {
+
+	// Test that three env variables are set
+	appVersion := os.Getenv("APP_VERSION")
+	require.True(t, appVersion != "", "APP_VERSION env var not set")
+	dockerRegistry := os.Getenv("DOCKER_REGISTRY")
+	require.True(t, dockerRegistry != "", "DOCKER_REGISTRY env var not set")
+	kubeconfig := os.Getenv("KUBECONFIG")
+	require.True(t, kubeconfig != "", "KUBECONFIG evn var not set")
+
+	// remove old cluster if it still exists
+	args := []string{
+		"delete",
+		"crdbclusters.crdb.cockroachlabs.com",
+		"cockroachdb",
+	}
+
+	process.ExecJUnit("oc", args, os.Environ())
+
+	// remove old pvc's from last run
+	args = []string{
+		"-n", "test",
+		"delete",
+		"pvc",
+		"-l",
+		"app.kubernetes.io/name=cockroachdb",
+	}
+
+	process.ExecJUnit("oc", args, os.Environ())
+
+	// remove test namespace
+	args = []string{
+		"delete",
+		"namespace",
+		"test",
+	}
+
+	process.ExecJUnit("oc", args, os.Environ())
+
+	// remove crds
+	args = []string{
+		"delete",
+		"crd",
+		"crdbclusters.crdb.cockroachlabs.com",
+	}
+
+	process.ExecJUnit("oc", args, os.Environ())
+
+	// create a new project/namespace for testing
+	args = []string{
+		"new-project",
+		"test",
+	}
+
+	// we cannot do a require.NoError on process, because oc tries to modify
+	// the kubeconfig file, and with bazel you cannot do this
+	process.ExecJUnit("oc", args, os.Environ())
+
+	// create a YAMLTemplate for the go templating
+	yaml := &YAMLTemplate{
+		AppVersion:     appVersion,
+		DockerRegistry: dockerRegistry,
+	}
+
+	// create some YAML files
+	catalogByte := parseTmpl(t, catalogSource, yaml)
+	subByte := parseTmpl(t, subscription, yaml)
+
+	catalogFilename := writeFile(t, catalogByte)
+	subFilename := writeFile(t, subByte)
+
+	var b bytes.Buffer
+	b.WriteString(operatorGroup)
+
+	operatorFilename := writeFile(t, b.Bytes())
+
+	files := []string{catalogFilename, subFilename, operatorFilename}
+
+	// install YAML files which install the operator
+	for _, f := range files {
+		fmt.Println(f)
+		args := []string{"delete", "-f", f}
```

Comment:
The files all have a `nonce` suffix, I doubt this will make a difference, maybe a `defer` to clean up at the end?

Reply:
If we fail with a err will the defer still run in the test framework?